internal/service/directconnect/router_configuration_data_source.go

```diff
@@ -0,0 +1,152 @@
+							Computed: true,
+						},
+						"vendor": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"xslt_template_name": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"xslt_template_name_for_mac_sec": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+			"router_type_identifier": {
+				Type: schema.TypeString,
+				// even though the API Reference shows this as optional, the API call will fail without this argument
+				Required: true,
+			},
+			"virtual_interface_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"virtual_interface_name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+const (
+	DSNameRouterConfiguration = "Router Configuration Data Source"
+)
+
+func dataSourceRouterConfigurationRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conn := meta.(*conns.AWSClient).DirectConnectConn
+
+	routerTypeIdentifier := d.Get("router_type_identifier").(string)
+	virtualInterfaceId := d.Get("virtual_interface_id").(string)
+
+	out, err := findRouterConfigurationByTypeAndVif(conn, routerTypeIdentifier, virtualInterfaceId)
+	if err != nil {
+		return create.DiagError(names.DirectConnect, create.ErrActionReading, DSNameRouterConfiguration, virtualInterfaceId, err)
+	}
+
+	d.SetId(fmt.Sprintf("%s:%s", virtualInterfaceId, routerTypeIdentifier))
+
+	d.Set("customer_router_config", out.CustomerRouterConfig)
+	if router_type_out := out.Router.RouterTypeIdentifier; router_type_out != nil {
+		d.Set("router_type_identifier", router_type_out)
+	}
+	d.Set("virtual_interface_id", out.VirtualInterfaceId)
+	d.Set("virtual_interface_name", out.VirtualInterfaceName)
+
+	if err := d.Set("router", flattenRouter(out.Router)); err != nil {
+		return create.DiagError(names.DirectConnect, create.ErrActionSetting, DSNameRouterConfiguration, d.Id(), err)
+	}
+
+	return nil
+}
+
+func findRouterConfigurationByTypeAndVif(conn *directconnect.DirectConnect, routerTypeIdentifier string, virtualInterfaceId string) (*directconnect.DescribeRouterConfigurationOutput, error) {
+	input := &directconnect.DescribeRouterConfigurationInput{
+		RouterTypeIdentifier: aws.String(routerTypeIdentifier),
+		VirtualInterfaceId:   aws.String(virtualInterfaceId),
+	}
+
+	output, err := conn.DescribeRouterConfiguration(input)
+
+	if err != nil {
+		return nil, err
+	}
+
+	if output == nil {
+		return nil, tfresource.NewEmptyResultError(input)
+	}
+
+	return output, nil
+}
+
+func flattenRouter(apiObject *directconnect.RouterType) []interface{} {
+	tfMap := map[string]interface{}{}
+
+	if v := apiObject.Platform; v != nil {
+		tfMap["platform"] = v
+	}
+
+	if v := apiObject.RouterTypeIdentifier; v != nil {
+		tfMap["router_type_identifier"] = v
+	}
+
+	if v := apiObject.Software; v != nil {
+		tfMap["software"] = v
+	}
+
+	if v := apiObject.Vendor; v != nil {
+		tfMap["vendor"] = v
```

Comment:
We can use an in-built function to return the value of the string value that is passed

```suggestion
		tfMap["vendor"] = aws.StringValue(v)